src/en/manhwaxxl/src/eu/kanade/tachiyomi/extension/en/manhwaxxl/ManhwaXXL.kt

```diff
@@ -85,42 +90,59 @@ class ManhwaXXL : ParsedHttpSource() {
     override fun searchMangaNextPageSelector() = popularMangaNextPageSelector()
 
     override fun mangaDetailsParse(document: Document) = SManga.create().apply {
-        val statusBadge = document.selectFirst("span.card-title i")?.classNames() ?: emptySet()
-
-        title = document.selectFirst("span.card-title h1")!!.text()
-        author = document.selectFirst("div:has(> i.fa-user)")?.ownText()
-        description = document.selectFirst("div.manga-info")?.text()
-        genre = document.select("ul.post-categories li").joinToString { it.text() }
-        status = when {
-            statusBadge.contains("fa-circle-check") -> SManga.COMPLETED
-            statusBadge.contains("fa-rotate") -> SManga.ONGOING
+        author = document.selectFirst("div.comic-details__item_links span")?.ownText()
+        description = document.selectFirst("section#post-decription h2 + p")?.text()
+        genre = document.select("div.comic-details__item a[href*=genre]").joinToString { it.text() }
+        status = when (document.selectFirst("div.card h5")?.text()) {
+            "Completed" -> SManga.ONGOING
+            "Ongoing" -> SManga.COMPLETED
             else -> SManga.UNKNOWN
         }
-        thumbnail_url = document.selectFirst("div.card div.manga-avatar img")?.absUrl("src")
     }
 
     // Manga details page have paginated chapter list. We sacrifice `date_upload`
     // but we save a bunch of calls, since each page is like 12 chapters.
     override fun chapterListParse(response: Response): List<SChapter> {
         val detailsDocument = response.asJsoup()
-        val firstChapter = detailsDocument.selectFirst("ul.chapters-list li.item-chapter a")?.absUrl("href")
-            ?: return emptyList()
-        val document = client.newCall(GET(firstChapter, headers)).execute().asJsoup()
+        val script = detailsDocument.selectFirst("script:containsData(post_id)")?.data()
+            ?: throw java.lang.Exception("Failed to get chapter id")
```

Comment:
import

Reply:
Sorry @AwkwardPeak7 didn't know what you mean by import

Reply:
`import java.lang.Exception` at the top so the code here can be:
```suggestion
            ?: throw Exception("Failed to get chapter id")
```